hybridse/src/udf/default_udf_library.cc

```diff
@@ -310,6 +310,47 @@ struct DistinctCountDef {
     };
 };
 
+template <typename T>
+struct MedianDef {
+    using ArgT = typename DataTypeTrait<T>::CCallArgType;
+    using VectorT = std::vector<T>;
+
+    void operator()(UdafRegistryHelper& helper) {
+        std::string suffix = ".opaque_vector_" + DataTypeTrait<T>::to_string();
+        helper.templates<Nullable<double>, Opaque<VectorT>, Nullable<T>>()
+            .init("median_init" + suffix, MedianDef::Init)
+            .update("median_update" + suffix, MedianDef::Update)
+            .output("meadin_output" + suffix, reinterpret_cast<void*>(MedianDef::Output), true);
+    }
+
+    static void Init(VectorT* addr) { new (addr) VectorT(); }
+
+    static VectorT* Update(VectorT* vec, T value, bool is_null) {
+        if (!is_null) {
+            vec->push_back(value);
+        }
+        return vec;
+    }
+
+    static void Output(VectorT* vec, double* ret, bool* is_null) {
+        if (vec->empty()) {
+            *is_null = true;
+        } else {
+            *is_null = false;
+
+            std::sort(vec->begin(), vec->end());
+            size_t mid = vec->size() / 2;
+            if (vec->size() % 2 == 0) {
+                *ret = (vec->at(mid - 1) + vec->at(mid)) / 2.0;
+            } else {
+                *ret = vec->at(mid);
+            }
+        }
+        vec->clear();
+        vec->~VectorT();
```

Comment:
I guess it is ok to just `delete` ?

Reply:
> I guess it is ok to just `delete` ?

This is what I did in the first place, but it leads to double free in testing. So I referred to the code in `DistinctCountDef` and `TopKDef`

https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/hybridse/src/udf/default_udf_library.cc#L292

https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/hybridse/src/udf/containers.h#L87